File: SDS/test_second/level1_2.java
import java.util.*;
import java.io.*;

public class level1_2 {

  public static void main(String[] args) {
    Scanner scan = new Scanner(System.in);
    int t, n, m, tot;
    int[] color = new int[51];
    int[] car = new int[1000];
    boolean[] isTarget = new boolean[51];

    t = scan.nextInt();
    for (int test_case = 1; test_case <= t; ++test_case) {
      n = scan.nextInt();
      m = scan.nextInt();
      tot = 0;

      for (int i = 1; i <= m; ++i) {
        color[i] = scan.nextInt();
        if (color[i] > 0) isTarget[i] = true;
        else isTarget[i] = false;
        tot += color[i]; 
      }
      for (int i = 0; i < n; ++i) {
        car[i] = scan.nextInt();
      }

      boolean isFound = false;
      int left = 0, right = 0;
      while (right < n && left <= right) {
        if (isTarget[car[right]]) {
          if (color[car[right]] > 0) {
            --tot;
            --color[car[right]];
            ++right;
            if (tot == 0) {
              isFound = true;
              break;
            }
          } else {
            while (car[left] != car[right]) {
              ++tot;
              ++color[car[left]];
              ++left;
            }
            ++left; ++right;
          }
        } else {
          while (left < right) {
            ++tot;
            ++color[car[left]];
            ++left;
          }
          ++left; ++right;
        }
      }

      System.out.print("#" + test_case + " ");
      if (isFound) System.out.println((left + 1));
      else System.out.println(0);
    }
  }
}

File: Other/SDS/test_second/level1_2.cpp
#include <iostream>
using namespace std;

int main(int argc, char *argv[]) {
  cin.tie(0);
  ios_base::sync_with_stdio(false);

  int T, N, M, color[51], car[1000], tot;
  bool isTarget[51];
  cin >> T;
  for (int test_case = 1; test_case <= T; ++test_case) {
    cin >> N >> M;
    tot = 0;
    for (int i = 1; i <= M; ++i) {
      cin >> color[i];
      if (color[i]) isTarget[i] = true;
      else isTarget[i] = false;
      tot += color[i];
    }
    for (int i = 0; i < N; ++i) {
      cin >> car[i];
    }

    bool isFound = false;
    int left = 0, right = 0;
    while (right < N && left <= right) {
      if (isTarget[car[right]]) {
        if (color[car[right]]) {
          --tot;
          --color[car[right]];
          ++right;
          if (!tot) {
            isFound = true;
            break;
          }
        } else {
          while (car[left] != car[right]) {
            ++tot;
            ++color[car[left]];
            ++left;
          }
          ++left; ++right;
        }
      } else {
        while (left < right) {
          ++tot;
          ++color[car[left]];
          ++left;
        }
        ++left; ++right;
      }
    }

    cout << '#' << test_case << ' ';
    if (isFound) cout << (left + 1) << '\n';
    else cout << 0 << '\n';
  }

  return 0;
}